src/EnergyPlus/WaterCoils.cc

```diff
@@ -2621,23 +2623,35 @@ namespace WaterCoils {
                     WaterCoil(CoilNum).DesInletAirHumRat = TempSize; // coil report
                     WaterCoil(CoilNum).InletAirHumRat = TempSize;
-                    TempSize = AutoSize; // these data are initially 0, set to autosize to receive a result from RequestSizing
-                    RequestSizing(state, CompType, CompName, HeatingAirflowUASizing, SizingString, TempSize, bPRINT, RoutineName);
-                    // WaterCoil( CoilNum ).InletAirMassFlowRate = TempSize;
                     WaterCoil(CoilNum).DesAirVolFlowRate = DataAirFlowUsedForSizing;                // coil report
                     WaterCoil(CoilNum).InletAirMassFlowRate = DataAirFlowUsedForSizing * StdRhoAir; // this is stiil volume flow!
                 } else {
-                    TempSize = AutoSize; // these data are initially 0, set to autosize to receive a result from RequestSizing
-                    RequestSizing(state, CompType, CompName, HeatingWaterDesAirInletTempSizing, SizingString, TempSize, bPRINT, RoutineName);
-                    WaterCoil(CoilNum).InletAirTemp = TempSize;
+                    HeatingWaterDesAirInletTempSizer sizer;
+                    sizer.initializeWithinEP(state, DataHVACGlobals::cAllCoilTypes(DataHVACGlobals::Coil_HeatingWater), WaterCoil(CoilNum).Name, false);
+                    AutoSizingResultType result = sizer.size(state, DataSizing::AutoSize);
+                    WaterCoil(CoilNum).InletAirTemp = sizer.autoSizedValue;
+                    coilSelectionReportObj->setCoilEntAirTemp(CompName, CompType, WaterCoil(CoilNum).InletAirTemp, CurSysNum, CurZoneEqNum);
+                    if (result != AutoSizingResultType::NoError) {
+                        ShowSevereError("Developer Error: autosizing of water heating coil design air inlet temperature failed.");
+                        ShowContinueError("Occurs in water heating coil object= " + WaterCoil(CoilNum).Name);
+                        ErrorsFound = true;
+                    }
+
                     TempSize = AutoSize; // these data are initially 0, set to autosize to receive a result from RequestSizing
                     RequestSizing(state, CompType, CompName, HeatingWaterDesAirInletHumRatSizing, SizingString, TempSize, bPRINT, RoutineName);
                     WaterCoil(CoilNum).DesInletAirHumRat = TempSize; // coil report
                     WaterCoil(CoilNum).InletAirHumRat = TempSize;
-                    TempSize = AutoSize; // these data are initially 0, set to autosize to receive a result from RequestSizing
-                    RequestSizing(state, CompType, CompName, HeatingAirflowUASizing, SizingString, TempSize, bPRINT, RoutineName);
-                    WaterCoil(CoilNum).DesAirMassFlowRate = TempSize; // coil report
-                    WaterCoil(CoilNum).InletAirMassFlowRate = TempSize;
+
+                    HeatingAirflowUASizer sizer2;
```

Comment:
How to delete sizer so we don't have to use sizer2, sizer 3, etc.

Reply:
I think we actually _should_ use more than one, and I kinda think we should make them member variables on the appropriate struct:

```
struct WaterCoilWhatever {
   ...
   HeatingAirflowUASizer heatingFlowSizer;
   CoolingAirflowUASizer coolingFlowSizer;
   ...
}
```

That way we always have access to the sizing information that occurred for that component whenever we need it.

Reply:
So there will be a lot of these in each component model. And do they have to be unique across component models?

Reply:
Fair point about there being a lot of them.  I guess a single water coil would need to hold all the possible sizing options for each autosizable component.  Let's think about a utopian world for a moment, where all these sizers are in a beautiful class structure like this:

```
struct BaseSizer {
  virtual size() = 0;
}

struct BaseHeatingFlowSizer : BaseSizer {
  ...all the things common to heating flow sizing...
}

struct HeatingAirflowUASizer : BaseHeatingFlowSizer {
  ...specifics about sizing heating airflow via UA...
  void size() override;
}

struct HeatingAirflowType2 : BaseHeatingFlowSizer {
  ...specifics about sizing heating airflow via type 2...
  void size() override;
}

struct HeatingAirflowType3 : BaseHeatingFlowSizer {
  ...specifics about sizing heating airflow via type 3...
  void size() override;
}
```

I believe your point here is that WaterCoils would need to hold instances of every one of these types:

```
struct WaterCoilWhatever {
  HeatingAirflowUASizer heatingFlowSizerUA;
  HeatingAirflowType2 heatingFlowSizerType2;
  HeatingAirflowType3 heatingFlowSizerType3;
}
```

...even though just one would be used as a time.  Well, in the utopian world, we could do:

```
struct WaterCoilWhatever {
  BaseHeatingFlowSizer * heatingFlowSizer;
}
```

Note that this sizer is an instance of the heating flow base class, so it can hold any of the derived heating flow types, we don't have to keep multiple types just for the same input field.  The end goal will be 1 sizer object per autosizable field.

I keep saying this is "utopian" because it feels like we are a ways off, but not really.  As long as we keep this in mind as we proceed, we will be able to get to this point pretty darn easily once these things are converted.

For now, maybe we keep these as local variables and just use them where we need them.  We'll worry about this as a next step later.